src/EditorFeatures/Core.Wpf/Preview/PreviewFactoryService.cs

```diff
@@ -445,8 +497,21 @@ public Task<object> CreateChangedDocumentPreviewViewAsync(Document oldDocument,
                 leftWorkspace, rightWorkspace, zoomLevel, cancellationToken);
         }
 
-        public Task<object> CreateChangedAdditionalDocumentPreviewViewAsync(TextDocument oldDocument, TextDocument newDocument, double zoomLevel, CancellationToken cancellationToken)
+        // NOTE: We are only sharing this code between additional documents and analyzer config documents,
+        // which are essentially plain text documents. Regular source documents need special handling
+        // and hence have a different implementation.
+        private Task<object> CreateChangedAdditionalOrAnalyzerConfigDocumentPreviewViewAsync(
+            TextDocument oldDocument,
```

Comment:
IDocumentService might be good candidate to make each different kind of document to have document specific behavior.